Source/ablastr/fields/PoissonSolver.H

```diff
@@ -264,17 +299,14 @@ computePhi (amrex::Vector<amrex::MultiFab*> const & rho,
 #ifdef AMREX_USE_OMP
 #pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
 #endif
-            for (amrex::MFIter mfi(*phi[lev+1],amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi)
-            {
-                amrex::Array4<amrex::Real> const& phi_fp_arr = phi[lev+1]->array(mfi);
-                amrex::Array4<amrex::Real> const& phi_cp_arr = phi_cp.array(mfi);
-
-                amrex::Box const& b = mfi.tilebox(phi[lev+1]->ixType().toIntVect());
-                amrex::ParallelFor( b,
-                                    [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
-                {
-                    amrex::mf_nodebilin_interp(i, j, k, 0, phi_fp_arr, 0, phi_cp_arr, 0, refratio);
-                });
+            for (amrex::MFIter mfi(*phi[lev + 1], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi) {
+                amrex::Array4<amrex::Real> const phi_fp_arr = phi[lev + 1]->array(mfi);
+                amrex::Array4<amrex::Real const> const phi_cp_arr = phi_cp.array(mfi);
+
+                details::PoissonInterpCPtoFP const interp(phi_fp_arr, phi_cp_arr, refratio);
```

Comment:
I intentionally copy the Array4's and `IntVect` here, which should not be an issue...
(we used `=` before as well in the lambda and all things are const-ifyied where possible)